app/services/proxy_status.rb

```diff
@@ -0,0 +1,55 @@
+class ProxyStatus
```

Comment:
:clap: - 1 minor nitpick, this looks like a model, not a service, shouldn't it be on /models?